### ISSUE TITLE
Rid use of non-portable gnat specific code.

### ISSUE THREAD

**Issue by charlie5** [charlie5/aShell#2]
Mostly to do with process & command argument handling.

**Comment by mgrojo** [comment]
Is this the reason why the crate specifically depends on gcc 11.2.4? I'm thinking in using your crate, but I'm afraid of limiting the availability of my crate by this requirement. Could it at least the requirement be reduced, like depending on any GCC 11 version?

**Comment by charlie5** [comment]
hi @mgrojo 

Ashell should be buildable with any GCC version >= 11.2.4 and with 'florist_blady' crate version >= 6.0.0.

I'm currently in the process of updating the ashell alire.toml file to reflect this.

Thanks for your interest.

**Comment by mgrojo** [comment]
Thanks for the quick response.

By the way, maybe you can shed some light to why a simple crate using withing your crate fails due to warnings being treated as errors. You can see the simple test reproducer in https://github.com/alire-project/alire/issues/1386#issuecomment-1722322512

I searched for some switch in your repository that activates the `-gnatwe` switch and found nothing. So the question is where that behaviour is coming from.

~In my case, I suppose the GCC version was coming from Ubuntu's gnat-10. After updating to gnat-11, I cannot longer reproduce the issue. Is then GCC 10 the one to blame for warnings being treated as errors without asking for it?~

Edited: sorry, it is still failing, I forgot to do `alr build` after `alr with ashell`.

**Comment by charlie5** [comment]
@mgrojo

Yes, I've seen this before in florist_blady project, which treats warnings as errors. The warnings are due to newer GCC versions checking for more unnecessary 'with's. I submitted a pull-request to fix these to blady's florist github repo, which was accepted. I am guessing he has not as yet made a new alire release with these changes.

It's quite late here and I'm a bit tired. I'll investigate properly tomorrow and let you know the result.

**Comment by mgrojo** [comment]
It's a strange problem, because only withing florist_blady, the problem is not reproduced, so I cannot understand where is `-gnatwe` added (or equivalent, maybe `gprbuild -we`). In any case, I'd expected one of those switches to appear in the crate directory, doing a recursive grep, but it doesn't find anything.

**Comment by mgrojo** [comment]
I know now where is the problem in the grep searching, those warning switches can be joined together, and looking for `-gnatw.*e` it finds the problem:

```
alire/cache/dependencies/florist_blady_6.0.0_3bc476cc/florist.gpr:            for Default_Switches ("Ada") use ("-g", "-gnatw.eH.YD", "-gnatyg", "-Wall",
alire/cache/dependencies/florist_blady_6.0.0_3bc476cc/gnatsocks/gnatsocks.gpr:            for Default_Switches ("Ada") use ("-g", "-gnatwae", "-gnatyg");
```
So I will open an issue in florist_blady suggesting removing that switch for the general profile used by the crate, otherwise any future warning raised by a compiler will make the crate to fail. This is good for the library developer, but not for the mere user.
